framework/application/application.cpp

```diff
@@ -117,37 +117,65 @@ bool Application::PlaySingleFrame()
     return success;
 }
 
-bool Application::RegisterWindow(decode::Window* window)
+void Application::ProcessEvents(bool wait_for_input)
 {
-    assert(window != nullptr);
-
-    if (std::find(windows_.begin(), windows_.end(), window) != windows_.end())
+    if (wsi_context_)
     {
-        GFXRECON_LOG_INFO("A window was registered with the application more than once");
-        return false;
+        wsi_context_->ProcessEvents(wait_for_input);
     }
-
-    windows_.push_back(window);
-
-    return true;
 }
 
-bool Application::UnregisterWindow(decode::Window* window)
+void Application::InitializeWsiContext(const char* surfaceExtensionName, void* pPlatformSpecificData)
 {
-    assert(window != nullptr);
-
-    auto pos = std::find(windows_.begin(), windows_.end(), window);
-
-    if (pos == windows_.end())
+    if (!wsi_context_)
     {
-        GFXRECON_LOG_INFO(
-            "A remove window request was made for an window that was never registered with the application");
-        return false;
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+        if (!strcmp(surfaceExtensionName, VK_KHR_WIN32_SURFACE_EXTENSION_NAME))
```

Comment:
There is a GFXR utility function for string compare `gfxrecon::util::platform::StringCompare`.